pkg/controller/reconciler/deployment_reconciler.go

```diff
@@ -124,9 +124,9 @@ func (r *DeploymentReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 		}
 	}
 
-	// If the Deployment is being deleted **or** if Radius is not enabled, then we should
+	// If the Deployment is being deleted **or** if Radius is no longer enabled, then we should
 	// clean up any Radius state.
-	if deployment.DeletionTimestamp != nil || annotations.Configuration == nil {
+	if deployment.DeletionTimestamp != nil || (annotations.Configuration == nil && annotations.Status != nil) {
```

Comment:
Logic bug was causing this code to process all deployments, not just those where Radius was being disabled.